packages/editor/src/client/routes/polls/pollEditView.tsx

```diff
@@ -0,0 +1,256 @@
+  useEffect(() => {
+    if (data?.poll) {
+      setPoll(data.poll)
+    } else {
+      setPoll(undefined)
+    }
+  }, [data])
+
+  useEffect(() => {
+    if (updateData?.updatePoll) {
+      setPoll(updateData.updatePoll)
+    } else {
+      setPoll(undefined)
+    }
+  }, [updateData])
+
+  /**
+   * Form validation model
+   */
+  const {StringType} = Schema.Types
+  const validationModel = Schema.Model({
+    question: StringType().isRequired(t('pollEditView.questionRequired'))
+  })
+
+  /**
+   * FUNCTIONS
+   */
+  async function saveOrUpdate(): Promise<void> {
+    if (!poll) {
+      return
+    }
+    const opensAt = poll.opensAt ? new Date(poll.opensAt).toISOString() : null
+    const closedAt = poll.closedAt ? new Date(poll.closedAt).toISOString() : null
+    const externalSources = poll.externalVoteSources?.map((voteSource: PollExternalVoteSource) => ({
+      ...voteSource,
+      __typename: undefined,
+      voteAmounts: voteSource.voteAmounts?.map((voteAmount: PollExternalVote) => ({
+        id: voteAmount.id,
+        amount: voteAmount.amount
+      }))
+    }))
+
+    await updatePoll({
+      variables: {
+        pollId: poll.id,
+        question: poll.question,
+        opensAt,
+        closedAt,
+        answers: poll.answers?.map(answer => {
+          return {
+            id: answer.id,
+            answer: answer.answer
+          }
+        }),
+        externalVoteSources: externalSources || []
+      }
+    })
+
+    if (close) {
+      navigate(closePath)
+    }
+  }
+
+  return (
+    <>
+      <Form
+        onSubmit={validationPassed => validationPassed && saveOrUpdate()}
+        model={validationModel}
+        fluid
+        disabled={loading}
+        formValue={{question: poll?.question}}>
+        <FlexboxGrid>
+          {/* model title */}
+          <FlexboxGrid.Item colspan={24}>
+            <ModelTitle
+              loading={loading}
+              title={poll?.question || t('pollList.noQuestion')}
+              loadingTitle={t('pollEditView.loadingTitle')}
+              saveBtnTitle={t('pollEditView.saveTitle')}
+              saveAndCloseBtnTitle={t('pollEditView.saveAndCloseTitle')}
+              closePath={closePath}
+              setCloseFn={setClose}
+            />
+          </FlexboxGrid.Item>
+
+          {/* content */}
+          <FlexboxGrid.Item colspan={12}>
+            <Row>
+              {/* question */}
+              <Col xs={24}>
+                <Panel header={t('pollEditView.questionPanelHeader')} bordered>
+                  <Form.Group controlId="question">
+                    <Form.Control
+                      name="question"
+                      placeholder={t('pollEditView.toBeOrNotToBe')}
+                      value={poll?.question || ''}
+                      onChange={(value: string) => {
+                        if (!poll) {
+                          return
+                        }
+                        setPoll({...poll, question: value})
```

Comment:
setState's inside of event handlers in React 17 are batched (in React 18 all setStates), thus this can lead to a bug as `poll` could be stale.

```suggestion
                        setPoll(p => ({...p, question: value}))
```

Would fix this issue.